appserver/web/web-core/src/main/java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -1031,6 +1043,27 @@ public int getPort() {
         return connector.getPort();
     }
 
+    /**
+     * Method to validate invalid characters on header values based on the specification RFC-9110
+     * @param req represents the Request that contains the Header values
+     * @return boolean false if any of the evaluated characters is available in any of the header values, true if the headers 
+     * don't contain any of the problematic characters
+     */
+    private boolean validateHeaderValues(final org.glassfish.grizzly.http.server.Request req) {
```

Comment:
Isn't this too late though? Shouldn'r Grizzly already reject an invalid Request? Based on original report it is indeed the case and Grizzly shouldn't even pass the request to Coyote.

Reply:
yeah at the beginning I thought to add it to the Grizzly side, after checking with the team we thought that Coyote side would be fine. @Pandrex247  any suggestion?